### DISCUSSION TITLE
Extracting the node which failed during tree execution

### DISCUSSION THREAD

**Discussion by Alexferlisi** [BehaviorTree/BehaviorTree.CPP#697] [Q&A]
Hello, i am using the library for my master thesis, in my code once the tree is created and ticked, if it returns failure i would like to know which tree node failed during execution, any suggestions?

Something like extracting each node of the tree and executing it one by one but as far as i know it seems not possible.

**Comment by facontidavide** [comment]
You can use the loggers. Also check TreeObserver
